spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -45,66 +45,51 @@
 @ConditionalOnKubernetesAndConfigEnabled
 @ConditionalOnClass(EndpointAutoConfiguration.class)
 @AutoConfigureAfter({ InfoEndpointAutoConfiguration.class, RefreshEndpointAutoConfiguration.class,
-		RefreshAutoConfiguration.class })
```

Comment:
I'll start this PR by tagging numbers in the order, you should start from (1) and go up - this will be easier to review, imho. It's a little bit involved, I have to admit. 

(1) So we had a class `ConfigReloadAutoConfiguration` in commons. It contained an inner class: `ConfigReloadAutoConfigurationBeans`.

`ConfigReloadAutoConfiguration` had no beans declared on its own, which, to me, means that we can take the conditionals from `ConfigReloadAutoConfigurationBeans` and place them on `ConfigReloadAutoConfiguration`, and as such drop `ConfigReloadAutoConfigurationBeans`. 

This simplifies this common class quite a lot (there are slight more simplifications in it in the next PR)